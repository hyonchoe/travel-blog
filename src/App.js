import React from 'react'
import { Switch, Route, Router, Link, useLocation } from 'react-router-dom'
import { Layout, Menu } from 'antd';

import history from './history'
import Home from './Home.js'
import EditTrip from './EditTrip.js'
import './App.css'

const { Header, Content, Footer } = Layout

class App extends React.Component {
    state = {
        trips: [],
        /**
         * {
         *  title: 'asdf',
         *  date: 'asdf',
         *  details: 'asdf',
         * },
         * ...
         */

         editTripIndex: null,
    }
    
    handleSubmit = trip => {
      this.setState({
          trips: [...this.state.trips, trip] 
        })
      history.push('/')
    }
    handleDeleteTrip = index => {
        this.setState({
            trips: this.state.trips.filter((trip, i) => {
                return i !== index
            })
        })
    }
    handleEditTrip = index => {
        this.setState({
            editTripIndex: index,
        })
        history.push('/addTrip')
    }
    handleUpdate = (updatedTrip, index) => {
        const clone = this.state.trips.slice()
        clone[index] = updatedTrip
        this.setState({
            trips: clone,
            editTripIndex: null,
        })
        history.push('/')
    }
    handleCancel = () => {
        history.push('/')
    }
    
    render() {
        const trips = this.state.trips
        const tripEditIndex = this.state.editTripIndex
        let tripToEdit = (tripEditIndex !== null) ? trips[tripEditIndex] : null

        return (
            <div className="appContainer">
                <Router history={history}>
                    <Layout className="appLayout">
                        <Header className="appHeader">
                            <div className="logo" />
                            <TopMenus />
                        </Header>

                        <Content className="appContent" >
                                    <Switch>
                                        <Route path="/" exact>
                                            <Home
                                                tripData={trips}
                                                deleteTrip={this.handleDeleteTrip}
                                                editTrip={this.handleEditTrip} />
                                        </Route>                
                                        <Route path="/addTrip">
                                            <EditTrip
                                                editTrip={tripToEdit}
                                                editTripIndex={tripEditIndex}
                                                handleUpdate={this.handleUpdate}
                                                handleSubmit={this.handleSubmit}
                                                handleCancel={this.handleCancel}
                                                />
                                        </Route>
                                    </Switch>
                        </Content>

                        <Footer className="appFooter">
                            Work in progress
                        </Footer>
                    </Layout>
                </Router>
            </div>
        )
    }
  }

const TopMenus = () =>{
    let location = useLocation()
    return (
        <Menu theme="dark" mode="horizontal" selectedKeys={[location.pathname]}>
            <Menu.Item key="/">
                <Link to="/">My Trips</Link>
            </Menu.Item>
            <Menu.Item key="/addTrip">
                <Link to="/addTrip">
                    Add/Edit Trip
                </Link>
            </Menu.Item>
        </Menu> 
    )   
}

export default App