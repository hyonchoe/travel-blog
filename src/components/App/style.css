.appContainer {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
}

.appContent {
    flex: 1;
    padding: 8px 50px;
}

.appFooter {
    text-align: center;
}

.footerWord {
    margin: 0px 5px;
}