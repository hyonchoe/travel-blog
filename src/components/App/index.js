import React from 'react'
import { Switch, Route, Router } from 'react-router-dom'
import { Layout } from 'antd'
import history from '../../services/history'
import DisplayTrips from '../DisplayTrips'
import EditTrip from '../EditTrip'
import PrivateRoute from '../PrivateRoute'
import NavBar from '../NavBar'
import Error404 from '../Error404'
import './style.css'

const { Header, Content, Footer } = Layout

class App extends React.Component {
    state = {
        tripToEdit: null,
    }
    
    componentDidMount() {}

    handleEditTrip = (trip) => {
        this.setState({
            tripToEdit: trip
        })
    }
    clearEditTrip = () => {
        this.setState({
            tripToEdit: null
        })
    }
    
    render() {
        const tripToEdit = this.state.tripToEdit

        return (
            <div className="appContainer">
                <Router history={history}>
                    <Layout className="appLayout">
                        <Header className="appHeader">
                            <NavBar />
                        </Header>

                        <Content className="appContent" >
                            <Switch>
                                <Route path="/" exact>
                                    <DisplayTrips
                                        showMyTrips={false}
                                        editTrip={this.handleEditTrip} />
                                </Route>
                                <PrivateRoute
                                    path="/myTrips"
                                    component={DisplayTrips}
                                    showMyTrips={true}
                                    editTrip={this.handleEditTrip} />
                                <PrivateRoute 
                                    path="/addTrip"
                                    component={EditTrip}
                                    editTrip={tripToEdit}
                                    clearEditTrip={this.clearEditTrip} />
                                <Route component={Error404} />
                            </Switch>
                        </Content>

                        <Footer className="appFooter">
                            <span className="footerWord">Travel</span>
                            <span className="footerWord">Explore</span>
                            <span className="footerWord">Learn</span>
                            <span className="footerWord">Remember</span>
                        </Footer>
                    </Layout>
                </Router>
            </div>
        )
    }
  }

export default App