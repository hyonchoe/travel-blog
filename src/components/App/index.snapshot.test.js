/**
 * Snapshot test for App component.
 *
 * Run by running 'npm test' in command line.
 */

import React from 'react'
import { shallow } from 'enzyme'
import App from './'

describe('App', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(
            <App />
        )
        expect(wrapper).toMatchSnapshot()
    })
})