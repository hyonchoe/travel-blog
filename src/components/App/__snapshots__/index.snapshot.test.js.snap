// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="appContainer"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Layout
      className="appLayout"
    >
      <Header
        className="appHeader"
      >
        <NavBar />
      </Header>
      <Content
        className="appContent"
      >
        <Switch>
          <Route
            exact={true}
            path="/"
          >
            <DisplayTrips
              editTrip={[Function]}
              showMyTrips={false}
            />
          </Route>
          <PrivateRoute
            component={[Function]}
            editTrip={[Function]}
            path="/myTrips"
            showMyTrips={true}
          />
          <PrivateRoute
            clearEditTrip={[Function]}
            component={[Function]}
            editTrip={null}
            path="/addTrip"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </Content>
      <Footer
        className="appFooter"
      >
        <span
          className="footerWord"
        >
          Travel
        </span>
        <span
          className="footerWord"
        >
          Explore
        </span>
        <span
          className="footerWord"
        >
          Learn
        </span>
        <span
          className="footerWord"
        >
          Remember
        </span>
      </Footer>
    </Layout>
  </Router>
</div>
`;
